### DISCUSSION TITLE
Change search to be portable

### DISCUSSION THREAD

**Discussion by teamcrisis** [thuliteio/doks#125] [Ideas]
Should search be changed to use portable `where` ?
[https://gohugo.io/functions/where/#mainsections](https://gohugo.io/functions/where/#mainsections)

I was trying to index more than just the docs section. Changing to `portable` would allow the user to specify which sections they would want indexed within the `config`.

Updating:
https://github.com/h-enk/doks/blob/master/assets/js/index.js#L86

To:
`{{ $pages := where site.RegularPages "Type" "in" site.Params.mainSections }}`

Then adding this to `config.toml`:
```
[params]
  mainSections = ["blog", "docs"]
```

**Comment by h-enk** [comment]
See also: https://github.com/h-enk/doks/discussions/120, https://github.com/h-enk/doks/issues/130

**Comment by dacort** [comment]
Thanks for the sample code @teamcrisis - I agree this would be really useful. I'm using doks as a site for my personal projects, so also added that as a custom archetype and needed to search across it.
